packages/view-types/statistical-plots/src/CellSetSizesPlotSubscriber.js

```diff
@@ -67,15 +72,160 @@ export function CellSetSizesPlotSubscriber(props) {
     [cellSets, additionalCellSets],
   );
 
-  // From the cell sets hierarchy and the list of selected cell sets,
-  // generate the array of set sizes data points for the bar plot.
-  const data = useMemo(() => (mergedCellSets && cellSetSelection && cellSetColor
-    ? treeToSetSizesBySetNames(mergedCellSets, cellSetSelection, cellSetColor, theme)
-    : []
-  ), [mergedCellSets, cellSetSelection, cellSetColor, theme]);
+  const getPaths = (node, currentPath = [], paths = []) => {
```

Comment:
Can these functions (`getPaths`, `filterPaths`, `findLongest`, `getNewHierarchy`, `convertSubarrayElements` and `getData`) be refactored into a new JS file containing utility functions, and then add unit tests and JSDoc comments for them?

Reply:
Yes, good point. I have put them in a new file called set-path-utils and created unit tests for them. Can you take a look and let me know what you think?